kdump-utils.spec

```diff
@@ -56,6 +56,10 @@ servicelog_notify --remove --command=/usr/lib/kdump/kdump-migrate-action.sh 2>/d
 servicelog_notify --add --command=/usr/lib/kdump/kdump-migrate-action.sh --match='refcode="#MIGRATE" and serviceable=0' --type=EVENT --method=pairs_stdin >/dev/null
 %endif
 
+# RPM scriptlet should always return 0. Otherwise it can break
```

Comment:
No, I don't think scripts should always exit 0.

In image-based deployments `%post` scripts are run on a build server. I think the fundamental problem here is that the `%post` script is trying to do something interacting with ppc64le hardware (?) in `servicelog_notify`. That operation ultimately needs to be in. something like a systemd unit or triggered as part of kdump's existing system unit.

I think a better change would have been to just add `|| true` after the `servicelog_notify` in the short term.

But the reason RPM scripts should not *always* exit 0 is precisely because we *do* want to capture actual errors in them at build time! Let's say the script is generating some data that expects to be read.

If we get say an I/O error or a transient ENOSPC, we don't want to stumble on and generate a broken image.

What I really need to do is argue for a global switch in RPM that would make errors fatal by default that we can turn on for image builds.

Reply:
Thanks for letting me know bootc prefers to not let RPM scriptlet always return 0 and suggesting to execute servicelog_notify elsewhere! Your reply also prompts me to learn about how bootc handles RPm script and help me realize the error should not be ignored. So I've filed https://issues.redhat.com/browse/RHEL-71502 to track this problem. 

Btw, I reached the conclusion that "RPM scriptlets should return 0" by reading the following texts in [Scriptlets :: Fedora Docs](https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_syntax) which I think needs to be adapted for bootc, 

> All scriptlets MUST exit with the zero exit status. Because RPM in its default configuration does not execute shell scriptlets with the -e argument to the shell, excluding explicit exit calls (frowned upon with a non-zero argument!), the exit status of the last command in a scriptlet determines its exit status. Most commands in the snippets in this document have a “|| :” appended to them, which is a generic trick to force the zero exit status for those commands whether they worked or not. Usually the most important bit is to apply this to the last command executed in a scriptlet, or to add a separate command such as plain “:” or “exit 0” as the last one in a scriptlet. Note that depending on the case, other error checking/prevention measures may be more appropriate.

> Non-zero exit codes from scriptlets can break installs/upgrades/erases such that no further actions will be taken for that package in a transaction (see [Ordering](https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#ordering)), which may for example prevent an old version of a package from being erased on upgrades, leaving behind duplicate rpmdb entries and possibly stale, unowned files on the filesystem. There are some cases where letting the transaction to proceed when some things in scriptlets failed may result in partially broken setup. It is however often limited to that package only whereas letting a transaction to proceed with some packages dropped out on the fly is more likely to result in broader system wide problems.